js/src/jupyter-leaflet.js

```diff
@@ -344,6 +345,41 @@ var LeafletVideoOverlayView = LeafletRasterLayerView.extend({
     },
 });
 
+var LeafletVelocityView = LeafletLayerView.extend({
+    create_obj: function () {
+        var options = this.get_options();
+        var data = this.model.get('data');
+        options.data = JSON.parse(data);
+        this.obj = L.velocityLayer(
+            options
+        );
+    },
+
+    model_events: function () {
+        LeafletVelocityView.__super__.model_events.apply(this, arguments);
+        this.listenTo(this.model, 'change:data', function () {
+            options = this.get_options();
+            data = this.model.get('data');
+            options.data = JSON.parse(data);
+            this.map_view.obj.removeLayer(this.obj);
+            this.obj = L.velocityLayer(options);
+            this.map_view.obj.addLayer(this.obj);
+        }, this);
+        var key;
+        var o = this.model.get('options');
+        for (var i=0; i<o.length; i++) {
+            key = o[i];
+            this.listenTo(this.model, 'change:' + key, function () {
+                options = this.get_options();
+                data = this.model.get('data');
+                options.data = JSON.parse(data);
+                this.map_view.obj.removeLayer(this.obj);
+                this.obj = L.velocityLayer(options);
+                this.map_view.obj.addLayer(this.obj);
```

Comment:
If you use `L.setOptions(this.obj, options)` instead of removing and adding back the layer, I guess it should work, but I'm really not sure for this one

Reply:
Yes it works, thanks!